### ISSUE TITLE
Implement cluster wide Pub/Sub framework

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#20320]
We currently have a cluster-wide mechanism to invalidate cache entries but that is the only cluster wide communication we use.  In our various CacheTransport providers, we enable them by using a pub/sub framework but we don't expose that framework for use by other parts of dotCMS.  We need to create a pub/sub framework that can be used for cross node communications.

The first phase of this is to get a POC framework in place and use it to implement a new CacheTransport mechanism that does not rely on 3rd party pub/sub libs like Hazelcast.  Our current crop of CacheTransports have a bug of disrupting a cluster's service when nodes come in and out and the cluster needs to be rewired, preventing it from being a good solution for auto-scalling. 

We need a cache transport mechanism that will allow for seamless autoscaling as nodes are added or removed from a cluster.  In the first iteration, I'd like to implement this transport using PostgreSQL's native pub/sub capabilities.  As we move forward, we can leverage different PubSub tooling such as Redis, or even reimplement Hazelcast to provide cluster wide capabilities.

**Comment by dsilvam** [comment]
PR ee:  https://github.com/dotCMS/enterprise/pull/804

**Comment by bryanboza** [comment]
This is causing a continuous NPE in the log 

Here the error: https://gist.github.com/bryanboza/18969f29eb835726eef22176e3b1c27d

**Comment by wezell** [comment]
Bryan, this is happening because you have not updated your db connection pool on your test server to use Hikari and the `db.properties` file

**Comment by dsilvam** [comment]
**Passed Internal QA**. Tested on a dotcms cluster with 3 nodes, using docker and specifying the cache transport via environment variable in the docker-compose: `"DOT_CACHE_INVALIDATION_TRANSPORT_CLASS": 'com.dotcms.cache.transport.PubSubCacheTransport'`. 

Results: 
* Network tab showing correct number of nodes and correct statuses for cache and ES
* Increasing or decreasing the number of nodes properly reflected on network tab
* Cache invalidations properly taking place on the different nodes.

**Comment by wezell** [comment]
Note to QA @bryanboza we need to also test that this still works:
```
CACHE_INVALIDATION_TRANSPORT_CLASS=com.dotcms.cache.transport.HazelcastCacheTransportEmbedded
```

**Comment by bryanboza** [comment]
Fixed, tested on release-21.06 // Postgres // FF
